File: Testing/Stdlib Test Code/test_string_methods.py
from dataclasses import dataclass, field
def strlen(s): return len(s)
def strstr(s, sub): idx = s.find(sub); return s[idx:] if idx >= 0 else None

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def str_len(s):
    return strlen(s)

def str_contains(s, sub):
    found = strstr(s, sub)
    return (found != None)

def main():
    s = 'Hello World'
    len = len(s)
    print(len)
    has = str_contains(s, 'World')
    print(has)
    return 0


if __name__ == "__main__":
    main()

File: Testing/Stdlib Test Code/test_string_methods.js
function strlen(s) { return s.length; }
function strstr(s, sub) { const i = s.indexOf(sub); return i >= 0 ? s.slice(i) : null; }


function str_len(s) {
    return strlen(s);
}

function str_contains(s, sub) {
    let found = strstr(s, sub);
    return (found !== null);
}

function main() {
    let s = "Hello World";
    let len = str_len(s);
    console.log(len);
    let has = str_contains(s, "World");
    console.log(has);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
